test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp

```diff
@@ -0,0 +1,150 @@
+                } Cases[] = {{0, 0, 0}, {1, 0, 1}, {0, 1, 1},   {1, 1, 1},
+                             {2, 3, 1}, {2, 4, 2}, {36, 17, 1}, {36, 18, 18}};
+                for (auto TC : Cases)
+                {
+                    { // Test with two signed types
+                        using Output = oneapi::dpl::common_type_t<S1, S2>;
+                        out[0] &= test0<S1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, -TC.y, TC.expect);
+                    }
+
+                    { // test with two unsigned types
+                        using Output = oneapi::dpl::common_type_t<U1, U2>;
+                        out[0] &= test0<U1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, U1, Output>(TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S1, U2>;
+                        out[0] &= test0<S1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, U2, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S2, U1>;
+                        out[0] &= test0<S2, U1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, U1, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                }
+                { //  LWG#2837
+                    auto res = oneapi::dpl::gcd(static_cast<std::int64_t>(1234), INT32_MIN);
+                    out[0] &= (res == 2);
+                }
+            });
+        });
+    }
+
+    if (res)
+        std::cout << "pass\n";
+    else
+        std::cout << "fail\n";
+}
+
+class KernelTest1;
+class KernelTest2;
+class KernelTest3;
+class KernelTest4;
+class KernelTest5;
+class KernelTest6;
+class KernelTest7;
+class KernelTest8;
+class KernelTest9;
+class KernelTest10;
+class KernelTest11;
+class KernelTest12;
+class KernelTest13;
+class KernelTest14;
+class KernelTest15;
+class KernelTest16;
+class KernelTest17;
```

Comment:
I think you don't need so much unique names. You can use data type (passed to the `do_test`) as a kernel name or part of the kernel name.

Reply:
Done.

Reply:
Sorry, I meant something different. I think we don't need such classes at all. You have
```
cgh.single_task<KernelTest>(...);
```
However, it could be 
```
template <typename T1, typename T2>
class KernelName;
...
cgh.single_task<KernelName<Input1, Input2>>(...);
```

Reply:
Many thanks. This makes code cleaner.